docker-compose.yml

```
@@ -0,0 +1,29 @@
+version: "3.9"
+name: sd-automatic
+services:
+  nvidia:
+    build:
+      context: .
+      args:
+        INSTALLDIR: "/webui"
+        RUN_UID: 1000
+    environment:
+      DATA_DIR: "./data"
+    volumes:
+      - ./data:/webui/data
```

Solves the [reinstall problem noted here](https://github.com/vladmandic/automatic/pull/1239#issuecomment-1622026536).

```suggestion
      - ./data:/webui/data
      - ./venv/lib:/webui/venv/lib
```

---

Even with the above, i still see the following on re-create but i'm not sure where they are coming from:
```
Downloading (…)olve/main/vocab.json: 100% 961k/961k [00:00<00:00, 34.7MB/s]
Downloading (…)olve/main/merges.txt: 100% 525k/525k [00:00<00:00, 43.1MB/s]
Downloading (…)cial_tokens_map.json: 100% 389/389 [00:00<00:00, 2.82MB/s]
Downloading (…)okenizer_config.json: 100% 905/905 [00:00<00:00, 3.26MB/s]
Downloading (…)lve/main/config.json: 100% 4.52k/4.52k [00:00<00:00, 13.2MB/s]
```

---

It would be more clean to declare those folders as `VOLUME`s in the Dockerfile as well. Then you could even leave out the bind mounts in the compose file so they will be created as anonymous volumes during launch.

---

/webui/venv/lib contains files that were installed when building the image, so I think it would require additional changes to mount it